packages/compass-aggregations/src/modules/pipeline-builder/pipeline-ai.ts

```diff
@@ -226,7 +255,12 @@ export const runAIPipelineGeneration = (
         throw new Error(emptyPipelineError);
       }
     } catch (err) {
-      logFailed((err as Error).message);
+      trackAndLogFailed({
```

Comment:
Wouldn't we want different log ids for this to be able to have a clearer picture in the logs of what happened and make it easier to find failure place in the code if needed? Why are we doing this through a generic method?

Reply:
I attached the `errorCode` and `errorName` to this log so it should be distinguishable where this is coming from. Using a generic method just to avoid repeating similar things and to make the `runAIPipelineGeneration` function more readable. Would you prefer each failure type has a separate code/handling function?

Reply:
Everything that ends in the extra attr of the log is not the easiest thing to read through, but if you feel like it's enough, feel free to keep it like that